homeassistant/components/buienradar/__init__.py

```diff
@@ -1 +1,162 @@
-"""The buienradar component."""
+"""The buienradar integration."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry
+
+from .const import (
+    CONF_COUNTRY,
+    CONF_DELTA,
+    CONF_DIMENSION,
+    CONF_TIMEFRAME,
+    DEFAULT_COUNTRY,
+    DEFAULT_DELTA,
+    DEFAULT_DIMENSION,
+    DEFAULT_TIMEFRAME,
+    DOMAIN,
+)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["weather", "camera", "sensor"]
+
+DATA_LISTENER = "listener"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the buienradar component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    weather_configs = _filter_domain_configs(config, "weather", DOMAIN)
+    sensor_configs = _filter_domain_configs(config, "sensor", DOMAIN)
+    camera_configs = _filter_domain_configs(config, "camera", DOMAIN)
+
+    _import_configs(hass, weather_configs, sensor_configs, camera_configs)
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up buienradar from a config entry."""
+    hass.data[DOMAIN][entry.entry_id] = {}
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    listener = entry.add_update_listener(async_update_options)
+    hass.data[DOMAIN][entry.entry_id][DATA_LISTENER] = listener
```

Comment:
After rebase, can be replaced by:

```suggestion
    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
    listener = entry.add_update_listener(async_update_options)
    hass.data[DOMAIN][entry.entry_id] = {DATA_LISTENER: listener}
```